src/Artsy/Router/RouterLink.tsx

```diff
@@ -14,7 +15,11 @@ import React, { useContext } from "react"
  */
 
 export const RouterLink: React.FC<LinkProps> = ({ to, children, ...props }) => {
-  const isRouterContext = !!useContext(RouterContext)
+  const context = useContext(RouterContext)
+  const routes = get(context, c => c.router.matcher.routeConfig, [])
+  const isSupportedInRouter = !!get(context, c =>
+    c.router.matcher.matchRoutes(routes, to)
+  )
```

Comment:
We pull in the new babel features to enable `foo?.bar?.baz` and `?? []` checks. Support just landed in the latest TypeScript version, for the type-checking side of things.